specs/electra/beacon-chain.md

```diff
@@ -615,6 +624,27 @@ def get_attesting_indices(state: BeaconState, attestation: Attestation) -> Set[V
     return output
 ```
 
+#### `get_activation_churn_consumption`
+
+```python
+def get_activation_churn_consumption(state: BeaconState, deposit: PendingDeposit) -> Gwei:
+    """
+    Return amount of activation churn consumed by the ``deposit``.
+    """
+    validator_pubkeys = [v.pubkey for v in state.validators]
+
+    if deposit.pubkey not in validator_pubkeys:
+        return deposit.amount
+    else:
+        validator_index = ValidatorIndex(validator_pubkeys.index(deposit.pubkey))
+        validator = state.validators[validator_index]
+        # Validator is exiting, do not consume the churn
```

Comment:
From @mkalinin this line is motivated by
- https://github.com/ethereum/consensus-specs/pull/3776

Reply:
The idea is to postpone deposit processing for an exiting validator until it has fully withdrawn. And then do the top-up without affecting WS period.